pkg/crc/config/combined_config.go

```diff
@@ -0,0 +1,135 @@
+package config
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/knadh/koanf"
+	jsonkoanf "github.com/knadh/koanf/parsers/json"
+	"github.com/knadh/koanf/providers/env"
+	"github.com/knadh/koanf/providers/file"
+	"github.com/knadh/koanf/providers/posflag"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+)
+
+type CombinedStorage struct {
+	// store env. variable
+	kEnv *koanf.Koanf
+	// store flag values
+	flagSet *pflag.FlagSet
+
+	// protect value stores
+	storeLock *sync.Mutex
+
+	configFile string
+}
+
+func NewCombinedStorage(configFile, envPrefix string) (*CombinedStorage, error) {
+	kBase := koanf.New(".")
```

Comment:
what is `kBase` supposed to mean? the `koanf` library's name by itself is not indicative enough of intent.